src/System.Linq/src/System/Linq/Take.cs

```diff
@@ -15,7 +15,8 @@ public static IEnumerable<TSource> Take<TSource>(this IEnumerable<TSource> sourc
             if (count <= 0) return new EmptyPartition<TSource>();
             IPartition<TSource> partition = source as IPartition<TSource>;
             if (partition != null) return partition.Take(count);
-            return TakeIterator(source, count);
+            IList<TSource> sourceList = source as IList<TSource>;
+            return sourceList != null ? new ListPartition<TSource>(sourceList, 0, count - 1) : TakeIterator(source, count);
```

Comment:
Nit: FWIW, in cases like this where there's already a bunch of cases like:
```C#
IInterface1 i1 = source as IInterface1;
if (i1 != null) return i1.Whatever();
IInterface1 i2 = source as IInterface2;
if (i2 != null) return i2.Whatever();
...
```
etc., I'd find the last one easier to understand as:
```C#
IInterfaceN iN = source as IInterfaceN;
if (iN != null) return iN.Whatever();
return Whatever();
```
rather than:
```C#
IInterfaceN iN = source as IInterfaceN;
return iN != null ? iN.Whatever() : Whatever();
```
Normally I like ```?:```, but in this case it breaks from the pattern used above it.